src/Microsoft.Identity.Client/ITokenCache.cs

```diff
@@ -27,34 +27,31 @@
 
 using Microsoft.Identity.Client.Cache;
 
-namespace Microsoft.Identity.Client.CacheV2
+namespace Microsoft.Identity.Client
 {
     /// <summary>
     /// This is the interface that implements the public access to cache operations.
-    /// With CacheV2, this should only be necessary if the caller is persisting
-    /// the cache in their own store, since this will provide the serialize/deserialize
-    /// and before/after notifications used in that scenario.
-    /// TODO: This interface should eventually be PUBLIC.
     /// </summary>
-    internal interface ITokenCache
+    public interface ITokenCache
     {
-        ///// <summary>
-        ///// 
-        ///// </summary>
-        ///// <param name="beforeAccess"></param>
-        //void SetBeforeAccess(TokenCacheNotification beforeAccess);
+#if !ANDROID_BUILDTIME && !iOS_BUILDTIME && !WINDOWS_APP_BUILDTIME
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="beforeAccess"></param>
+        void SetBeforeAccess(TokenCache.TokenCacheNotification beforeAccess);
 
-        ///// <summary>
-        ///// 
-        ///// </summary>
-        ///// <param name="afterAccess"></param>
-        //void SetAfterAccess(TokenCacheNotification afterAccess);
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="afterAccess"></param>
+        void SetAfterAccess(TokenCache.TokenCacheNotification afterAccess);
```

Comment:
Do we want to have a public nested type ? TokenCacheNotification? What's the pro?

Reply:
I'd rather not have a nested type, I'd rather have the delegate be in the namespace.  changing this will likely be breaking to existing code that uses these notifications.